File: web/client/src/app/model/account-transfer-export/smbc-koufuri-exporter.ts


import { ExporterBase } from 'src/app/model/account-transfer-export/exporter-base'
import { AccountTransferDetail } from 'src/app/model/account-transfer-detail.model'

export class SMBCKoufuriExporter extends ExporterBase
{
    public GetHeaderRecord():string{
        return this.JoinFixedData([
            this.GetStrField(1,  this.DivisionHeader),
            this.GetStrField(2,  91),
            this.GetStrField(1,  1),
            this.GetStrField(2,  "00"),
            this.GetNmbField(8,  this.PaymentAgency.consigneeCode), /* 右8桁切取 */
            this.GetStrField(40, this.Company.kana),
            this.GetStrField(4,  this.GetMMdd(this.DueAt)),
            this.GetNmbField(4,  this.PaymentAgency.bankCode),
            this.GetStrField(15, this.PaymentAgency.bankName),
            this.GetNmbField(3,  this.PaymentAgency.branchCode),
            this.GetStrField(15, this.PaymentAgency.branchName),
            this.GetStrField(1,  this.PaymentAgency.accountTypeId),
            this.GetNmbField(7,  this.PaymentAgency.accountNumber),
            this.GetStrField(17, this.Dummy())]);
    }

    public GetDataRecord(detail:AccountTransferDetail):string{
        return this.JoinFixedData([
            this.GetStrField(1,  this.DivisionData),
            this.GetNmbField(4,  detail.transferBankCode),
            this.GetStrField(15, detail.transferBankName),
            this.GetNmbField(3,  detail.transferBranchCode),
            this.GetStrField(15, detail.transferBranchName),
            this.GetStrField(4,  this.Dummy()),
            this.GetStrField(1,  detail.transferAccountTypeId),
            this.GetNmbField(7,  detail.transferAccountNumber),
            this.GetStrField(30, detail.transferAccountName),
            this.GetNmbField(10, parseInt(detail.billingAmount.toString())),
            this.GetStrField(1,  detail.transferNewCode),
            this.GetNmbField(8,  this.PaymentAgency.consigneeCode), /* 右8桁切取 */
            this.GetNmbField(12, detail.transferCustomerCode), /* 左12桁切取 検証実施済  */
            this.GetStrField(1,  0),
            this.GetStrField(8,  this.Dummy())]);
    }

    public GetTrailerRecord():string{
        return this.JoinFixedData([
            this.GetStrField(1,  this.DivisionTrailer),
            this.GetNmbField(6,  this.TotalCount),
            this.GetNmbField(12, parseInt(this.TotalAmount.toString())),
            this.GetNmbField(6,  0),
            this.GetNmbField(12, 0),
            this.GetNmbField(6,  0),
            this.GetNmbField(12, 0),
            this.GetStrField(65, this.Dummy())]);
    }

    public GetEndRecord():string{
        return this.JoinFixedData([
            this.GetStrField(1,   this.DivisionEnd),
            this.GetStrField(119, this.Dummy())]);
    }
}

File: src/Rac.VOne.AccountTransfer/Export/SMBCKoufuriExporter.cs
using Rac.VOne.Web.Models;

namespace Rac.VOne.AccountTransfer.Export
{
    public class SMBCKoufuriExporter : ExporterBase
    {
        public override string GetHeaderRecord() => Join(
            GetStrField( 1, DivisionHeader),
            GetStrField( 2, 91),
            GetStrField( 1, 1),
            GetStrField( 2, "00"),
            GetNmbField( 8, PaymentAgency.ConsigneeCode), /* 右8桁切取 */
            GetStrField(40, Company.Kana),
            GetStrField( 4, DueAt.ToString("MMdd")),
            GetNmbField( 4, PaymentAgency.BankCode),
            GetStrField(15, PaymentAgency.BankName),
            GetNmbField( 3, PaymentAgency.BranchCode),
            GetStrField(15, PaymentAgency.BranchName),
            GetStrField( 1, PaymentAgency.AccountTypeId),
            GetNmbField( 7, PaymentAgency.AccountNumber),
            GetStrField(17, Dummy()));
        public override string GetDataRecord(AccountTransferDetail detail) => Join(
            GetStrField( 1, DivisionData),
            GetNmbField( 4, detail.TransferBankCode),
            GetStrField(15, detail.TransferBankName),
            GetNmbField( 3, detail.TransferBranchCode),
            GetStrField(15, detail.TransferBranchName),
            GetStrField( 4, Dummy()),
            GetStrField( 1, detail.TransferAccountTypeId),
            GetNmbField( 7, detail.TransferAccountNumber),
            GetStrField(30, detail.TransferAccountName),
            GetNmbField(10, decimal.Truncate(detail.BillingAmount)),
            GetStrField( 1, detail.TransferNewCode),
            GetNmbField( 8, PaymentAgency.ConsigneeCode), /* 右8桁切取 */
            GetNmbField(12, detail.TransferCustomerCode), /* 左12桁切取 検証実施済  */
            GetStrField( 1, 0),
            GetStrField( 8, Dummy())
            );

        public override string GetTrailerRecord() => Join(
            GetStrField( 1, DivisionTrailer),
            GetNmbField( 6, TotalCount),
            GetNmbField(12, decimal.Truncate(TotalAmount)),
            GetNmbField( 6, 0),
            GetNmbField(12, 0),
            GetNmbField( 6, 0),
            GetNmbField(12, 0),
            GetStrField(65, Dummy()));
        public override string GetEndRecord() => Join(
            GetStrField(  1, DivisionEnd),
            GetStrField(119, Dummy()));
    }
}
